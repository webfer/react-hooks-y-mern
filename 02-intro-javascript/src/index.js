// file of work

const active = true;

// let msg = '';
//
// if (active) {
//   msg = 'active';
// } else {
//   msg = 'Unactive';
// }

//const msg = active ? 'Active' : 'unactive';
//const msg = active ? 'Active' : null;

//const msg = active && 'Active';
const msg = active && 'Active';
console.log(msg);
