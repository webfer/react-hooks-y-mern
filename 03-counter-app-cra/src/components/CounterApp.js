import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 5 }) => {
  const [counter, setCounter] = useState(value);

  // handleAdd
  const handleAdd = () => {
    setCounter(counter + 1); // This way is valid too
    // setCounter((c) => c + 1); // This way is valid too
  };

  const handleReduce = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleReduce}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
