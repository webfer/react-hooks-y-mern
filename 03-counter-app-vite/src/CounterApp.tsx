import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  //console.log("Render");
  // API .... FETCH ....
  // Al hacer un cambio en el estado se vuelve a renderizar todo el FC = Functional component

  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    //   console.log(event);
    //setCounter((c) => c + 1); // Otra fora de actualizar el state, cuando no se conoce el valor de referencia (counter)
    setCounter(counter + 1);
  };

  const handleDecrease = () => setCounter(counter - 1);

  const handleRset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAdd}> +1</button>
      <button onClick={handleDecrease}> -1</button>
      <button aria-label="btn-reset" onClick={handleRset}>
        {' '}
        Reset
      </button>
    </>
  );
};

CounterApp.prototypes = {
  value: PropTypes.number.isRequired,
};
