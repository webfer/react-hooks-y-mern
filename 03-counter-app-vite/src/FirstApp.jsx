import React from 'react';
import PropTypes from 'prop-types';

// const getSaludo = () => {
//   let saludo = {
//     name: "Fernando",
//     lastName: "Castro",
//   };
//   return saludo;
// };

export const FirstApp = ({
  title,
  name,
  subTitle,
  // subTitle = "Sub-titulo que pasa por defecto",
}) => {
  return (
    <>
      {/* <h1>Hello, {getSaludo().name + " " + getSaludo().lastName}</h1> */}
      <h1 data-testid="test-title">{title}</h1>
      <h2>Hello, {name}</h2>
      <h3>{subTitle}</h3>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: 'No hay nombre',
  subTitle: 'No hay subtitle',
  // title: "Este es el titulo",
};
