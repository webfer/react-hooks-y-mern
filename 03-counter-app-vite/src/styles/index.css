html,
body {
  background-color: #21232a;
  color: #fff;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  font-size: 1.3rem;
  margin-right: 5px;
  padding: 15px;
}
