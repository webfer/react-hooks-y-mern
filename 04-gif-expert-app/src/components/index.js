export * from './AddCategory';
export * from './GifExpertApp';
export * from './GifGrid';
export * from './GifItem';
