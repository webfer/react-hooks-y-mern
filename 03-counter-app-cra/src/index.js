import React from 'react';
import { createRoot } from 'react-dom/client';
// import PrimeraApp from './components/PrimeraApp';
import CounterApp from './components/CounterApp';

import './index.css';

const divRoot = document.getElementById('app');
const root = createRoot(divRoot);

root.render(<CounterApp value={10} />);
