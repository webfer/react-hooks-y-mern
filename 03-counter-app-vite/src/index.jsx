import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';

import './styles/index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp name="Fernando" title="Hola, soy Vegeta!" /> */}
    <CounterApp value={10} />
  </React.StrictMode>
);
