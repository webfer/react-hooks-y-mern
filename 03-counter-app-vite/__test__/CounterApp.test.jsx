import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Testing on CounterApp', () => {
  const initialValue = 10;
  test('CounterApp debe hacer match con el snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });
  test('CounterApp debe mostrar el valor inicail de 100', () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getAllByText(initialValue)).toBeTruthy();
    // expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain( '100'); // Other option
  });
  test('Debe incrementar con el boton +1', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('+1'));
    expect(screen.getByText('11')).toBeTruthy();
  });
  test('Debe decrementar con el boton -1', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('-1'));
    // screen.debug();
    expect(screen.getByText('9')).toBeTruthy();
  });
  test('Debe de funcionar el boton de reset', () => {
    render(<CounterApp value={'355'} />);
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByText('+1'));
    // fireEvent.click(screen.getByText('Reset'));
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
    expect(screen.getByText(355)).toBeTruthy();
  });
});
