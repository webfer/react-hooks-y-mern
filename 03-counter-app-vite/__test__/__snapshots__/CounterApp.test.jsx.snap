// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on CounterApp CounterApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
     +1
  </button>
  <button>
     -1
  </button>
  <button
    aria-label="btn-reset"
  >
     
    Reset
  </button>
</div>
`;
